EJB Runtime Statistics

spriadka opened the pull request:
This PR brings tests for viewing runtime EJB statistics.

spriadka commented:
There are still some failures in tests which need to be resolved before merging, it is ``MessageDrivenEJBTest.verifyExecutionTime``, ``MessageDrivenEJBTest.verifyInvocations``, ``MessageDrivenEJBTest.verifyPeakConcurrentInvocations``, ``SingletonEJBTest.verifyExecutionTime``, ``StatefulEJBTest.verifyPeakConcurrentInvocations``. As the amount of files in this PR is quite big, any tips/comments/advices are welcome

zebrik commented on tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/ejb/MessageDrivenEJBExecutionTime.java:
```diff
@@ -0,0 +1,36 @@
+    public void onMessage(Message message) {
+        TextMessage msg;
+        try {
+            Thread.sleep(5000);
```
It's quite long. How many time is this sleep called? Such long sleep appears 9 times in the code of this pull request.

zebrik commented on tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/MessageDrivenEJBTest.java:
```diff
@@ -0,0 +1,225 @@
+            ejbDeployment.getDeploymentName().indexOf(".war"));
+        HttpGet httpGet;
+        try {
+            URL url = new URL("http://0.0.0.0:8080/" + deploymentName + "/MessageDriven");
```
Hardcoded port. Maybe we can start thinking about adding the port offset to the testsuite configuration. Just note, not needed to be fixed immediately.

zebrik commented on tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/MessageDrivenEJBTest.java:
```diff
@@ -0,0 +1,225 @@
+            .verifyAttribute("run-as-role", EJBFixtures.ROLE_1);
+    }
+
+    private static void invoke(int numberOfInvocations, EJBDeployment ejbDeployment) {
```
Seems similar method is written more times. What about moving this method to some library class? WDYT?

zebrik commented on tests/basic/src/test/java/org/jboss/hal/testsuite/test/runtime/ejb/ejb/MessageDrivenEJB.java:
```diff
@@ -0,0 +1,35 @@
+            throw new RuntimeException(e);
+        }
+    }
+}
```
You can add a new line at the end of the file. Can be applied to more files in this pull request.

spriadka commented:
Thank you for the feedback @zebrik, I have moved the ``client.close()`` method higher in the abstraction, made the ``invoke(...)`` abstract and reduced the sleep times. I have added the hardcoded port thing for further work.
